### PR TITLE
Remove the unused `RecordBasedRepository.findRecords()` method

### PR DESCRIPTION
**Pull request by dmitrykuzmin**
This PR resolves issue https://github.com/SpineEventEngine/core-java/issues/792.

The method `RecordBasedRespository.findRecords()`, as it turns out, is used only in outdated Spine repositories (more specifically, [deployment](https://github.com/SpineEventEngine/deployment)). 

So it was deleted from the class.

Also there is a minor test refactoring as a side effect of initially writing test for the deleted method.

### CODE REVIEW

**Comment by dmitrykuzmin** [comment]
@armiol PTAL.

File: server/src/main/java/io/spine/server/entity/RecordBasedRepository.java [hunk 1]
```diff
@@ -317,6 +317,7 @@ protected E findOrCreate(I id) {
      * @param fieldMask the mask to apply for returned records
      * @return an iterator over the matching records
      */
+    @SuppressWarnings("WeakerAccess") // Used in the code outside of the repository.
```
Comment (armiol) [inline server/src/main/java/io/spine/server/entity/RecordBasedRepository.java]: Please clarify the usages.

**Comment by dmitrykuzmin** [comment]
@armiol PTAL again. I deleted the method.

**Comment by armiol** [comment]
@dmitrykuzmin @alexander-yevsyukov as discussed previously, this method was only used in an outdated version of `deployment` module. For now we delete this method from the public API. And we'll get back to `development` implementation in its good time.

@dmitrykuzmin LGTM.

**Pull request merged**